src/components/paginator-render/paginator-render.tsx

```diff
@@ -0,0 +1,668 @@
+
+  /**
+   * `true` to render the Items Per Page control, default = false;
+   */
+  @Prop() readonly showItemsPerPage: boolean = false;
+
+  /**
+   * Specifies the options in Items Per Page combo-box
+   */
+  @Prop() readonly itemsPerPageOptions: ComboBoxModel;
+
+  /**
+   * `true` to render the Items Per Page Info, default = false;
+   */
+  @Prop() readonly showItemsPerPageInfo: boolean = false;
+
+  /**
+   * `true` to render the navigation controls, default = true;
+   */
+  // eslint-disable-next-line @stencil-community/ban-default-true
+  @Prop() readonly showNavigationControls: boolean = true;
+
+  /**
+   * `true` to render the navigation Go To control, default = false;
+   */
+  @Prop() readonly showNavigationGoTo: boolean = false;
+
+  /**
+   * `true` to render the navigation First control, default = false;
+   */
+  @Prop() readonly showFirstControl: boolean = false;
+
+  /**
+   * `true` to render the navigation Last control, default = false;
+   */
+  @Prop() readonly showLastControl: boolean = false;
+
+  /**
+   * `true` to render the navigation Prev control, default = false;
+   */
+  @Prop() readonly showPrevControl: boolean = false;
+
+  /**
+   * `true` to render the navigation Next control, default = false;
+   */
+  @Prop() readonly showNextControl: boolean = false;
+
+  /**
+   * `true` to render the navigation Info, default = false;
+   */
+  @Prop() readonly showNavigationControlsInfo: boolean = false;
+
+  /**
+   * Specifies the order for the controls
+   */
+  @Prop() readonly order: ControlsOrder = {
+    itemsPerPage: 1,
+    itemsPerPageInfo: 2,
+    firstControl: 3,
+    prevControl: 4,
+    navigationControls: 5,
+    nextControl: 6,
+    lastControl: 7,
+    navigationGoTo: 8,
+    navigationControlsInfo: 9
+  };
+
+  /**
+   * Specifies the literals required in the control.
+   */
+  @Prop() readonly translations: PaginatorTranslations = {
+    text: {
+      goToButton: "Go",
+      itemsPerPage: "Items per page:",
+      of: "of ",
+      ellipsis: "...",
+      first: "First",
+      prev: "Prev",
+      next: "Next",
+      last: "Last",
+      unknownPages: "many",
+      unknownItems: "many",
+      showingItems: `Showing ${AMOUNT_ITEMS_IN_PAGE} of ${TOTAL_ITEMS} items`,
+      page: "Page:",
+      showingPage: `Showing ${ACTUAL_PAGE} of ${TOTAL_PAGES} pages`
+    }
+  };
+
+  /**
+   * Event fired when there is a new page selection.
+   */
+  @Event() pageChange: EventEmitter<number>;
+
+  /**
+   * Event fired when the amount of items per page changes.
+   */
+  @Event() itemsPerPageChange: EventEmitter<number>;
+
+  connectedCallback(): void {
+    this.hasHyperlink = Array.isArray(this.model);
+
+    if (this.hasHyperlink) {
+      this.totalPages = (this.model as PaginatorRenderHyperlinkModel).length;
+    } else {
+      this.totalPages =
+        (this.model as PaginatorRenderNumericModel)?.totalPages ?? -1;
+    }
+  }
+
+  #getPaginationRange() {
```

Comment:
Prefer arrow function.

```suggestion
  #getPaginationRange = () => {
```